src/phoenix/server/trace_handler.py

```diff
@@ -0,0 +1,53 @@
+import asyncio
+import gzip
+from typing import Protocol
+
+from google.protobuf.message import DecodeError
+from opentelemetry.proto.collector.trace.v1.trace_service_pb2 import (
+    ExportTraceServiceRequest,
+)
+from opentelemetry.proto.trace.v1.trace_pb2 import Span
+from starlette.endpoints import HTTPEndpoint
+from starlette.requests import Request
+from starlette.responses import Response
+from starlette.status import HTTP_415_UNSUPPORTED_MEDIA_TYPE, HTTP_422_UNPROCESSABLE_ENTITY
+
+
+class SupportsPutSpan(Protocol):
+    def put(self, span: Span) -> None:
+        ...
```

Comment:
Kinda confused by the indirection - we only have one traces object in core - do we really need this interface?

Reply:
It's meant to be a (generic) decoupling mechanism: 1. to highlight the fact that a queue is all you need, and 2. to prevent usage of any other features from the traces object.